pkg/trace/writer/trace.go

```diff
@@ -254,10 +256,8 @@ func (w *TraceWriter) flush() {
 
 	w.stats.BytesUncompressed.Add(int64(len(b)))
 
-	w.wg.Add(1)
-	go func() {
+	func() {
```

Comment:
Previously, launching a goroutine here meant that the memory consumption could grow unbounded.
If the sender channel was full the goroutines would just block waiting for room while new goroutines were launched.